docs/contributing/architecture/tests.md

```diff
@@ -31,7 +31,8 @@ packages/test-harness/src/config/init.lua
   -> TestHarnessRun() -> run() -> runAllTests() -> Mocha + packages/cursorless-neovim-e2e/src/suite/recorded.neovim.test.ts
 ```
 
-4. **Lua unit tests for neovim**: tests lua functions. They need to be executed in neovim. There is no such test for VSCode because we assume the functions provided by [VSCode APIs](https://code.visualstudio.com/api/references/vscode-api) have already been tested by Microsoft and work properly
```

Comment:
Just to note I removed this part about VSCode APIs because I think it's irrelevant. It almost implies we are testing neovim builtin APIs or something. The lua tests for cursorless.nvim are more analogous to the cursorless (non-action) unit tests imo.

Reply:
Yes, it is interesting. 

The reason I say the "lua helper functions" in neovim are similar to the "vscode APIs" in vscode is because we have to basically emulate the vscode APIs by combining lots of lua API calls. 

This is because Cursorless was built for VSCode in the first place so is very dependent on VSCode APIs architecture for now. It will probably change at some point and there may be "typescript unit tests for vscode" later (which would test combining lots of typescript calls, but specifically testing in the case of vscode, though it would indeed also be a way to test the lua calls in the case of neovim, and in that sense I agree they will be closer to the cursorless unit tests).

So ya, I think long term the lua tests for Cursorless neovim will be similar to the typescript unit tests for vscode. But we are not there yet I think.

Reply:
Curious what is your view on this @pokey?

Reply:
Thanks for the explanation. fwiw even with that explanation, imo, including what I deleted still doesn't make sense. If you want to highlight that neovim side needs custom lua functions that mimic vscode api and those need to be tested and so is why these tests are for, then I'd change the wording to just say that instead. saying we don't need to test vscode's api and linking to those api is still quite confusing (to me).

Reply:
Yes, I think it is better with old text removed. But I just thought I would explain. So fine with left as is.